client/src/components/hooks/useSquidMutation.jsx

```diff
@@ -0,0 +1,15 @@
+import { useMutation } from "react-query"
+
+import { ApiClient } from "../../backend/ApiClient"
+
+export const useSquidMutation = (squidQuery, setSubmitMessage) =>
+  useMutation((formPayload) => ApiClient.post("/squids", formPayload).then((resp) => resp.data), {
+    onSuccess: () => {
+      setSubmitMessage("Squid Added Successfully")
```

Comment:
rather than pass the explicit function in here. I think the hook should take a `onSuccess` callback.

```javascript
export const useSquidMutation = (squidQuery, onSuccess) => useMutation(() => {}, {onSuccess})
```